lib_nn/api/nn_layers.h

```diff
@@ -186,7 +186,7 @@ extern void pad_3_to_4_ref(int32_t outputs[], int8_t inputs[], uint32_t N_24, ui
  *
  * @returns  The inner product
  */
-extern void pad_3_to_4_asm(int32_t outputs[], int64_t inputs[], uint32_t N_24, uint32_t pad_val);
+extern void pad_3_to_4_asm(int8_t outputs[], int64_t inputs[], uint32_t N_24, uint32_t pad_val);
```

Comment:
Do we use int64_t[] for the input for alignment reasons?

Reply:
yes, I inherited it from Henk's code. I think strictly I should revert this line as it needs to be (and always is )32 bit aligned